.github/workflows/e2e.yml

```diff
@@ -42,6 +42,19 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-go-
 
+      # The default disk size of these runners is ~14GB, this is not enough to run the e2e tests.
+      # Cleanup the disk, see upstream discussion https://github.com/actions/runner-images/issues/2840.
+      - name: Cleanup Disk Space
+        run: |
+          echo "Before removing files:"
+          df -h
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "/usr/local/share/boost"
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          echo "After removing files:"
```

Comment:
I think we can do a lot more btw - in another project I use a custom action for this (https://github.com/mesosphere/kind-docker-image-automation/blob/tag-sync-automation/.github/workflows/build-kind-image.yaml#L27-L30) and we can do more if we need to later. Not a blocker, just in case we hit this again or this clean up isn't enough.